ocaml/xapi/cli_operations.ml

```diff
@@ -4340,12 +4340,22 @@ let update_upload fd printer rpc session_id params =
     let sr =
       if List.mem_assoc "sr-uuid" params
       then Client.SR.get_by_uuid rpc session_id (List.assoc "sr-uuid" params)
-      else Client.Pool.get_default_SR ~rpc ~session_id ~self:(List.hd pools)
-  in
+      else begin
+        let sr = Client.Pool.get_default_SR ~rpc ~session_id ~self:(List.hd pools) in
+        let ref_is_valid = Server_helpers.exec_with_new_task
```

Comment:
I see you've copied `let ref_is_valid ... in` from elsewhere in the file.
Is that worth factoring out into a function to be called from both places?
Then at this call site you could have just
```
if is_valid_ref sr then sr
else failwith ...
```